src/miner.cpp

```diff
@@ -146,6 +146,28 @@ bool CheckForDuplicatedSerials(const CTransaction& tx, const Consensus::Params&
     return true;
 }
 
+bool CreateCoinbaseTx(CBlock* pblock, const CScript& scriptPubKeyIn, CBlockIndex* pindexPrev)
+{
+    // Create coinbase tx
+    CMutableTransaction txNew;
+    txNew.vin.resize(1);
+    txNew.vin[0].prevout.SetNull();
+    txNew.vout.resize(1);
+    txNew.vout[0].scriptPubKey = scriptPubKeyIn;
+
+    //Masternode and general budget payments
+    FillBlockPayee(txNew, pindexPrev, false);
+
+    txNew.vin[0].scriptSig = CScript() << pindexPrev->nHeight + 1 << OP_0;
```

Comment:
This should be also on coinbases of PoS blocks.
But it seems that `IncrementExtraNonce` takes care of it. We can probably just remove it from here too.

Reply:
good catch. Added 94bcde9 refactoring it properly.